### PR TITLE
[DOCS] Moves Watcher troubleshooting page

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/stack-docs/pull/405

This PR moves a documentation file from the stack-docs repo to the elasticsearch repo, so it stays with the rest of the Watcher documentation.

It also updates the location of the Watcher troubleshooting page, since the "Troubleshooting" page in the Stack Overview is being retired.

### CODE REVIEW

**Review by jakelandis — APPROVED** [review]
LGTM

**Comment by mark-vieira** [comment]
This change looks to have caused some docs tests to fail in our intake builds. To unblock folks I've reverted this in all three branches for now. It looks like the one of the PR checks also failed for the same reason so we should probably try and ensure we are green across the board before merging.

https://scans.gradle.com/s/3iop543eu4yu6/tests/failed

**Pull request merged**
